### PR TITLE
feat: `HelmRepository` CRD support

### PR DESCRIPTION
**Pull request by maciaszczykm**
## Checklist

- [ ] If required, I have updated the Plural documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a meaningful title and summary to convey the impact of this PR to a user.

### CODE REVIEW

**Comment by michaeljguarino** [comment]
@maciaszczykm could you also add the `url` field to the `ServiceHelm` type in the operator too?  This will be needed for native helm support

**Comment by maciaszczykm** [comment]
@michaeljguarino Yes, sure. 

Should I update any resource to add refs to HelmRepository at this point?

**Comment by maciaszczykm** [comment]
@michaeljguarino Yes, I will do it. 

Should I update any resource to add refs to HelmRepository at this point?

**Comment by maciaszczykm** [comment]
@michaeljguarino Done in https://github.com/pluralsh/console/pull/1246/commits/7755c02726eb4a8a82d35587f2436ab268738625.

Should I update any resource to add refs to HelmRepository?

File: go/controller/internal/controller/helmrepository_controller.go [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+func (in *HelmRepositoryReconciler) Reconcile(ctx context.Context, req reconcile.Request) (_ reconcile.Result, retErr error) {
+	logger := log.FromContext(ctx)
+
+	helmRepository := new(v1alpha1.HelmRepository)
+	if err := in.Get(ctx, req.NamespacedName, helmRepository); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	utils.MarkCondition(helmRepository.SetCondition, v1alpha1.ReadyConditionType, v1.ConditionFalse, v1alpha1.ReadyConditionReason, "")
+
+	scope, err := NewHelmRepositoryScope(ctx, in.Client, helmRepository)
```
Comment (floreks) [inline go/controller/internal/controller/helmrepository_controller.go:63]: If there is no custom logic required for the scope you can use the generic scope (`NewDefaultScope`) same as stack definition and observability provider reconcilers.

File: go/controller/internal/controller/servicedeployment_controller.go [hunk 2]
```diff
@@ -351,9 +351,8 @@ func (r *ServiceReconciler) genServiceAttributes(ctx context.Context, service *v
 			attr.Helm.Values = values
 		}
 
-		if service.Spec.Helm.Chart != nil {
-			attr.Helm.Chart = service.Spec.Helm.Chart
-		}
+		attr.Helm.Chart = service.Spec.Helm.Chart
+		attr.Helm.URL = service.Spec.Helm.URL
```
Comment (floreks) [inline go/controller/internal/controller/servicedeployment_controller.go:355]: nit: could be moved to struct declaration (line 312).

**Review by michaeljguarino — APPROVED** [review]

**Pull request merged**